plugins/json/src/main/resources/struts-plugin.xml

```diff
@@ -5,8 +5,11 @@
 	"http://struts.apache.org/dtds/struts-2.5.dtd">
 
 <struts>
-    <bean class="org.apache.struts2.json.JSONWriter"/>
-    <bean class="org.apache.struts2.json.JSONUtil" static="true"/>
+    <bean type="org.apache.struts2.json.JSONWriter" name="json" class="org.apache.struts2.json.DefaultJSONWriter"
+          scope="prototype"/>
+    <constant name="struts.json.writer" value="json"/>
```

Comment:
hm... I think it's not needed, by using `struts` name above, it will be a default implementation

Reply:
Yes I see but it seems your above mechanism works in core only. I could not make it works:
With `@Inject(JSONConstants.JSON_WRITER)` I always get:
```text
No mapping found for dependency [type=org.apache.struts2.json.JSONWriter, name='struts.json.writer'] in public void org.apache.struts2.json.JSONUtil.setWriter(org.apache.struts2.json.JSONWriter).
```
independent to have or not have `<constant name="struts.json.writer" value="?" />`.

Without that but with name `struts` I get same message but looks for name `default`. With name `default` it works but I then could not override DefaultJSONWriter :)

I learnt my previous mechanism from convention plugin. If you have no idea about above errors? then I can keep it as is.

Reply:
... I change the name to struts, however; but keep injection by container as convention plugin does.

Reply:
Right... I forgot that isn't the core and we have no option to define aliases in plugins ... need to fix that :) Nice work 👍